### PR TITLE
Add root user to interactive users

### LINKED ISSUE #11699: file_permission_user_init_files is misaligned with DISA
#### Description of problem:
file_permission_user_init_files is misaligned with DISA

#### Details:
The SSG's rule hasn't found any files with wrong permissions.

DISA's rule has found these files:

```
/root/.bash_profile	regular	0	0	141	rw-r--r-- 
/root/.bash_logout	regular	0	0	18	rw-r--r-- 
/root/.bashrc	regular	0	0	429	rw-r--r-- 
/root/.cshrc	regular	0	0	100	rw-r--r-- 
/root/.tcshrc	regular	0	0	129	rw-r--r-- 
```

#### Outcome:
SSG result: pass
DISA result: fail

The issue is present in these test variants:
- oscap
- ansible
- anaconda

#### SCAP Security Guide Version:
Current upstream master as of 2024-03-12 as of HEAD cbbca44.

####  External Content's Version:
DISA STIG RHEL 9 V1R1

### PR DESCRIPTION
**Pull request by jan-cerny**
The rule file_permission_user_init_files now checks only dot files of users with UID greater than or equal 1000. But according to RHEL 9 STIG and CIS benchmarks it should check also the root user's dot files. This commit extends the rule to account also for the root user and adds a simple test scenario covering this situation.

Fixes #11699

### CODE REVIEW

**Review by Mab879 — APPROVED** [review]

**Comment by Mab879** [comment]
Tests pass locally

```
$ ./automatus.py rule --datastream ../build/ssg-rhel9-ds.xml --libvirt qemu:///system automatus_rhel9_4 file_permission_user_init_files
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/Developer/ComplianceAsCode/content/tests/logs/rule-custom-2024-03-19-1855/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_file_permission_user_init_files
INFO - Script all_permissions.fail.sh using profile (all) OK
INFO - Script correct_permissions.pass.sh using profile (all) OK
INFO - Script different_home_correct_permissions.pass.sh using profile (all) OK
INFO - Script different_home_wrong_permissions.fail.sh using profile (all) OK
INFO - Script lenient_permissions.fail.sh using profile (all) OK
INFO - Script stricter_permissions.pass.sh using profile (all) OK
INFO - Script lenient_permissions_root.fail.sh using profile (all) OK
$ ./automatus.py rule --datastream ../build/ssg-rhel9-ds.xml --libvirt qemu:///system automatus_rhel9_4 file_permission_user_init_files
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/Developer/ComplianceAsCode/content/tests/logs/rule-custom-2024-03-19-1855/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_file_permission_user_init_files
INFO - Script all_permissions.fail.sh using profile (all) OK
INFO - Script correct_permissions.pass.sh using profile (all) OK
INFO - Script different_home_correct_permissions.pass.sh using profile (all) OK
INFO - Script different_home_wrong_permissions.fail.sh using profile (all) OK
INFO - Script lenient_permissions.fail.sh using profile (all) OK
INFO - Script stricter_permissions.pass.sh using profile (all) OK
INFO - Script lenient_permissions_root.fail.sh using profile (all) OK
```

File: shared/macros/10-oval.jinja [hunk 1]
```diff
@@ -1157,6 +1157,17 @@ Generates the :code:`<affected>` tag for OVAL check using correct product platfo
   {{%- set ignored_users_list="(nobody|nfsnobody)" %}}
 
   <unix:password_object id="{{{ object_id }}}" version="1">
+    <set>
+      <object_reference>{{{ object_id }}}_root</object_reference>
+      <object_reference>{{{ object_id }}}_others</object_reference>
+    </set>
+  </unix:password_object>
+
+  <unix:password_object id="{{{ object_id }}}_root" version="1">
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply shared/macros/10-oval.jinja:1166]: yes already working on it

File: linux_os/guide/system/accounts/accounts-session/file_permission_user_init_files/bash/shared.sh [hunk 2]
```diff
@@ -6,9 +6,9 @@
 
 {{{ bash_instantiate_variables("var_user_initialization_files_regex") }}}
 
-readarray -t interactive_users < <(awk -F: '$3>={{{ uid_min }}}   {print $1}' /etc/passwd)
-readarray -t interactive_users_home < <(awk -F: '$3>={{{ uid_min }}}   {print $6}' /etc/passwd)
-readarray -t interactive_users_shell < <(awk -F: '$3>={{{ uid_min }}}   {print $7}' /etc/passwd)
+readarray -t interactive_users < <(awk -F: '$3==0 || $3>={{{ uid_min }}}   {print $1}' /etc/passwd)
```
Comment (Mab879) [inline linux_os/guide/system/accounts/accounts-session/file_permission_user_init_files/bash/shared.sh:9]: This always checks root do we want that?

**Review by Mab879 — COMMENTED** [review]
The scan has option for root, but the remediation doesn't. Is that expected.

File: linux_os/guide/system/accounts/accounts-session/file_permission_user_init_files/bash/shared.sh (hunk 2 above)
Comment (jan-cerny) [reply linux_os/guide/system/accounts/accounts-session/file_permission_user_init_files/bash/shared.sh:9]: I think yes - I think that is the point of the issue that the root's files were not evaluated and were not remediatied.

**Review by Mab879 — CHANGES_REQUESTED** [review]
The testing farm failure for RHEL 7 looks valid, please take look.

**Comment by jan-cerny** [comment]
The rule passes after first remediation and passes after second remediation but suddenly it fails in the final scan. I suspect something happens during the third reboot. I will try to identify the cause.

**Comment by jan-cerny** [comment]
The problem is that it conflicts with the rule rpm_verify_permissions. The root's dot files are owned by the rootfiles RPM package and they're shipped with different permissions. The first remediation changes the permissions in the code for rule file_permission_user_init_files, but that makes the rule rpm_verify_permissions fail and this fail is fixed during the second remediation and that makes the rule file_permission_user_init_files fail.

**Comment by jan-cerny** [comment]
We're trying to fix a RHEL 9 STIG misalignement. But the conflict with rpm_verify_permissions happens on RHEL 7. The reason is that RHEL 9 STIG doesn't contain rpm_verify_permissions. So we can solve this situation this way: We won't change rule file_permission_user_init_files. We will create a new rule that will be similar but will check the root files. We will replace the file_permission_user_init_files in RHEL 9 STIG with the new rule.  We will select the new rule only in RHEL 9 STIG. This way, we won't break other products. What do you think?

**Comment by Mab879** [comment]
> We're trying to fix a RHEL 9 STIG misalignement. But the conflict with rpm_verify_permissions happens on RHEL 7. The reason is that RHEL 9 STIG doesn't contain rpm_verify_permissions. So we can solve this situation this way: We won't change rule file_permission_user_init_files. We will create a new rule that will be similar but will check the root files. We will replace the file_permission_user_init_files in RHEL 9 STIG with the new rule. We will select the new rule only in RHEL 9 STIG. This way, we won't break other products. What do you think?

That seems like a fine solution.

**Comment by jan-cerny** [comment]
OK I will proceed with my proposal

**Review by Mab879 — APPROVED** [review]
LGTM.

Thanks

**Pull request merged**